Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs

```diff
@@ -149,8 +152,7 @@ public ISiteBuilder CreateBuilder(ITracer tracer, ILogger logger, IDeploymentSet
         private ISiteBuilder ResolveNonAspProject(string repositoryRoot, string projectPath, IDeploymentSettingsManager perDeploymentSettings)
         {
             string sourceProjectPath = projectPath ?? repositoryRoot;
-            // "FUNCTIONS_EXTENSION_VERSION" environment variable implies a functionApp
-            if (FunctionAppHelper.LooksLikeFunctionApp())
+            if (IsFunctionApp(sourceProjectPath))
```

Comment:
it used to be like this, I changed in the last commit, because it doesn't take a `projectPath` as argument like its siblings https://github.com/projectkudu/kudu/blob/5ab5fbcf795a24340b7b80ce694ad5dc66a907d1/Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs#L196-L199

but since we don't throw error when there's multiple deployable project
I think we need to check for "host.json" so the logic is more strict 
which bring it back to the old pattern, now it stand side by side as others:
https://github.com/projectkudu/kudu/blob/5da2e47f725799ed432c91cd644f7ba48d89be51/Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs#L189-L192